src/invariant/OrderBookIsNotCrossed.h

```diff
@@ -30,16 +30,16 @@ struct OfferEntryCmp
         {
             return true;
         }
-        else if (((a.flags & PASSIVE_FLAG) == 0) &&
-                 ((b.flags & PASSIVE_FLAG) != 0))
-        {
-            // We order non-passive offers before passive offers so that we can
-            // find any non-passive offer at a given price at the front of the
-            // list of offers at that price.
-            return true;
-        }
         else if (priceA == priceB)
         {
+            if (((a.flags & PASSIVE_FLAG) == 0) &&
```

Comment:
This still violates the asymmetry requirement.
```
L: Price = 1/1, Flags = 0, OfferID = 2
R: Price = 1/1, Flags = PASSIVE_FLAG, OfferID = 1
```

Then

- `L < R == true` because `price(L) == price(R)` and `(L.flags & PASSIVE_FLAG == 0) && (R.flags & PASSIVE_FLAG != 0) == true`
- `R < L == true` because `price(R) == price(L)` and `(R.flags & PASSIVE_FLAG == 0) && (L.flags & PASSIVE_FLAG != 0) == false` and `id(R) < id(L) == true`

Reply:
Good catch -- I should be checking the reverse condition on flags as well.  I've pushed a commit to do that.

While staring at it for a while to see what _else_ I might have messed up, I also decided that I like it better without the else-after-returns.  So I removed them in a separate commit.  There appears to me to be some legitimate dispute over else-after-return, but `clang-tidy`, for example, is unambiguous (https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html), as is Chromium (https://chromium.googlesource.com/chromium/src/+/refs/tags/73.0.3627.0/styleguide/c++/c++.md), and those two strike me as the most authoritative opinions that I've found so far.

Reply:
I've also now added your two examples in a new `TEST_CASE` specifically for `OfferEntryCmp`.